src/status_im/ui/screens/home/views/inner_item.cljs

```diff
@@ -101,22 +104,33 @@
 (defview home-list-chat-item-inner-view [{:keys [chat-id name color online
                                                  group-chat contacts public?
                                                  public-key unremovable? :as chat]}]
-  (letsubs [last-message [:get-last-message chat-id]]
-    (let [name (or (i18n/get-contact-translated chat-id :name name)
-                   (gfycat/generate-gfy public-key))]
-      [react/view styles/chat-container
-       [react/view styles/chat-icon-container
-        [chat-icon.screen/chat-icon-view-chat-list chat-id group-chat name color online]]
-       [react/view styles/chat-info-container
-        [react/view styles/item-upper-container
-         [chat-list-item-name name group-chat public? public-key]
-         (when last-message
-           [react/view styles/message-status-container
-            [message-status chat last-message]
-            [message-timestamp last-message]])]
-        [react/view styles/item-lower-container
-         [message-content-text last-message]
-         [unviewed-indicator chat-id]]]])))
+  (letsubs [last-message [:get-last-message chat-id]
+            swiped?      [:delete-swipe-position chat-id]]
+    (let [name                (or (i18n/get-contact-translated chat-id :name name)
+                                  (gfycat/generate-gfy public-key))
+          offset-x            (animation/create-value (if swiped? styles/delete-button-width 0))
+          swipe-pan-responder (responder/swipe-pan-responder offset-x styles/delete-button-width chat-id swiped?)
+          swipe-pan-handler   (responder/pan-handlers swipe-pan-responder)]
+      [react/view (merge {:style {}} swipe-pan-handler)
+       [react/animated-view {:style {:right offset-x}}
+        [react/touchable-highlight {:on-press #(re-frame/dispatch [:navigate-to-chat chat-id])}
+         [react/view styles/chat-container
+          [react/view styles/chat-icon-container
+           [chat-icon.screen/chat-icon-view-chat-list chat-id group-chat name color online]]
+          [react/view styles/chat-info-container
+           [react/view styles/item-upper-container
```

Comment:
Can we make this whole block more modular by extracting some function?

Reply:
@jeluard, thanks for reviewing. A idea to make it modular is to have a structure of:
home-list-item -> deletable wrapper -> chat-item/ browser-item.
Which is discussed with @flexsurfer in previous comment. 
This part in the I don't think need to be changed, because currently `chat-item` is different from `browser-item`, there is rarely common part except of this `deletable wrapper`. This structure now is good for comparison and further refactoring.

Reply:
My comment was not about having reusable parts but splitting it in smaller chunks to improve readability.

Looking at it in more details this is probably good enough.